### PR TITLE
add new features related with sidecarset

### PR DESCRIPTION
**Pull request by coldTea214**
### Ⅰ. Describe what this PR does

add new features related with sidecarset:
* support maxUnavailable parameter for upgrading sidecar
* support defining volumes in sidecarset

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.

pkg/controller/sidecarset/sidecarset_controller_test.go
pkg/webhook/default_server/pod/mutating/pod_mutating_test.go
pkg/webhook/default_server/sidecarset/mutating/sidecarset_mutating_test.go
pkg/webhook/default_server/sidecarset/validating/sidecarset_validating_test.go

### Ⅳ. Describe how to verify it

set spec.volumes in sidecarset, these volumes will be injected into matched pod
set spec.strategy.rollingupdate.maxUnavailable, this field will take effect when upgrading sidecar

### CODE REVIEW

File: pkg/apis/apps/v1alpha1/sidecarset_types.go [hunk 1]
```diff
@@ -29,15 +30,33 @@ type SidecarSetSpec struct {
...
 
 // SidecarContainer defines the container of Sidecar
 type SidecarContainer struct {
 	corev1.Container
 }
 
+// SidecarSetUpdateStrategy indicates the strategy that the SidecarSet
+// controller will use to perform updates. It includes any additional parameters
+// necessary to perform the update for the indicated strategy.
+type SidecarSetStrategy struct {
```
Comment (jian-he) [inline pkg/apis/apps/v1alpha1/sidecarset_types.go]: Call it SidecarSetUpdateStrategy?

File: pkg/webhook/default_server/sidecarset/mutating/sidecarset_create_update_handler.go [hunk 2]
```diff
@@ -57,10 +62,25 @@ type SidecarSetCreateHandler struct {
...
+	setSidecarSetStratety(&sidecarset.Spec.Strategy)
+
 	for i := range sidecarset.Spec.Containers {
 		setDefaultContainer(&sidecarset.Spec.Containers[i])
 	}
+
+	klog.V(3).Infof("sidecarset after mutating: %v", util.DumpJSON(sidecarset))
+}
+
+func setSidecarSetStratety(strategy *appsv1alpha1.SidecarSetStrategy) {
```
Comment (Fei-Guo) [inline pkg/webhook/default_server/sidecarset/mutating/sidecarset_create_update_handler.go]: Rename to setDefaultSidecarSetUpdateStrategy?

File: pkg/apis/apps/v1alpha1/sidecarset_types.go (hunk 1 above)
Comment (coldTea214) [reply pkg/apis/apps/v1alpha1/sidecarset_types.go]: done

File: pkg/webhook/default_server/sidecarset/mutating/sidecarset_create_update_handler.go (hunk 2 above)
Comment (coldTea214) [reply pkg/webhook/default_server/sidecarset/mutating/sidecarset_create_update_handler.go]: done

**Pull request merged**